pkg/apis/kubeflow.org/v2alpha1/trainingruntime_types.go

```diff
@@ -15,3 +15,173 @@ limitations under the License.
 */
 
 package v2alpha1
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	jobsetv1alpha2 "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+// ClusterTrainingRuntime represents a training runtime which can be referenced as part of
+// `trainingRuntimeRef` API in TrainJob. This resource is a cluster-scoped and can be referenced
+// by TrainJob that created in *any* namespace.
+type ClusterTrainingRuntime struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired ClusterTrainingRuntime.
+	Spec TrainingRuntimeSpec `json:"spec,omitempty"`
```

Comment:
Should the spec ever be empty for ClusterTrainingRUntime?

Reply:
Not really, but I noticed that for all Kubernetes APIs the `spec` is set with `omitempty`: https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L820.
@tenzen-y @kannon92 Any specific reason why we do this ?

Reply:
IIUC, in any case, the `spec` field is defined as an optional field in the Kubernetes. So, the optional TrainingRuntime spec would be better.

Reply:
TIL.